src/data/rat/cast.lean

```diff
@@ -276,3 +276,11 @@ end
 
 instance rat.subsingleton_ring_hom {R : Type*} [semiring R] : subsingleton (ℚ →+* R) :=
 ⟨ring_hom.ext_rat⟩
+
+lemma rat.forall {p : ℚ → Prop} : (∀ r, p r) ↔ ∀ a b : ℤ, 0 < b → p (a / b : ℚ) :=
```

Comment:
I put `rat.forall` in `cast.lean`, for I need all the coercion-lemma to work with tidy. I tried to put it in `basic.lean`, but it does not work. Any suggestion for a better file location?

Reply:
Proof without tidy

```lean
lemma rat.forall {p : ℚ → Prop} : (∀ r, p r) ↔ ∀ a b : ℤ, p (a / b : ℚ) :=
begin
  split,
  { intros h a b,
    exact h _ },
  { intros h q,
    have : q = q.num / (q.denom : ℤ),
    { simp [rat.div_num_denom] },
    rw this,
    exact h _ _ }
end
```

Reply:
also, you don't need `b > 0` :)

Reply:
Fixed these, thank you!